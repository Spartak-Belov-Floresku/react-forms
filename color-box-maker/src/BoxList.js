import { useState } from "react";
import { v4 as uuid } from 'uuid';

import Box from "./Box";
import NewBoxForm from "./NewBoxForm";


const BoxList = () => {

    const [boxes, setBoxs] = useState([]);

    const addBox = newBox => setBoxs(boxes => [...boxes, {...newBox, id : uuid()}]);

    const removeBox = divId => setBoxs(boxes.filter(div => div.id !== divId));
    
    return (
        <>
        <NewBoxForm addBox={addBox}/>
            <div>
                {boxes.lenght !== 0? boxes.map(({ id, color, width, height }) => <Box id={id} color={color} width={parseInt(width)} height={parseInt(height)} key={id} removeBox={removeBox} />):''}
            </div>
        </>
    )
}

export default BoxList;