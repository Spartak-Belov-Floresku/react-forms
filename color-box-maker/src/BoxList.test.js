import { render, fireEvent } from "@testing-library/react";
import BoxList from "./BoxList";

it("renders without crashing", () => {

    render(<BoxList />);

});

it("matches snapshot", () => {

    const { asFragment } = render(<BoxList />);
    expect(asFragment()).toMatchSnapshot();

});

if("should add new div and remove div", () => {

    const { queryByText } = render(<BoxList />);
    const addDIV = queryByText("Add DIV");

    expect(queryByText('X')).not.toBeInTheDocument();
    fireEvent.click(addDIV);
    expect(queryByText('X')).toBeInTheDocument();

    const removeDIV = queryByText('X');
    fireEvent.click(removeDIV);
    expect(queryByText('X')).not.toBeInTheDocument();

});