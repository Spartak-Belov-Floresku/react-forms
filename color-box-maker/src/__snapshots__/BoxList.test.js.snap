// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="color"
    >
      Color: 
    </label>
    <select
      id="color"
      name="color"
    >
      <option
        value="red"
      >
        Red
      </option>
      <option
        value="blue"
      >
        Blue
      </option>
      <option
        value="green"
      >
        Green
      </option>
      <option
        value="yellow"
      >
        Yellow
      </option>
    </select>
    <label
      for="width"
    >
      Width: 10 px 
    </label>
    <input
      id="width"
      max="200"
      min="10"
      name="width"
      type="range"
      value="10"
    />
    <label
      for="height"
    >
      Height: 10 px 
    </label>
    <input
      id="height"
      max="200"
      min="10"
      name="height"
      type="range"
      value="10"
    />
    <button>
      Add DIV
    </button>
  </form>
  <div />
</DocumentFragment>
`;
